### PR TITLE
Resolve 'Windows implementation'

### PR DESCRIPTION
**Pull request by leonx98**

### CODE REVIEW

File: desktop.ini [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[ViewState]
```
Comment (jagmit) [inline desktop.ini]: This file should be gitignored AFAIK

File: lib/src/app_handler.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import 'dart:typed_data';
+import 'package:meta/meta.dart';
+
+// Described an specific application which can open a file
+class AppHandler {
```
Comment (jagmit) [inline lib/src/app_handler.dart:6]: We should annotate this class as `@immutable` and definitely make all instance fields `final`

File: lib/src/app_handler.dart [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  String name;
+  String url;
+  Uint8List? icon;
+
+  @internal
+  AppHandler(this.name, this.url, {this.icon});
```
Comment (jagmit) [inline lib/src/app_handler.dart]: I personally prefer to not mix regular parameter with named ones. Flutter's SDK mostly also follows this style.

File: lib/src/app_handler.dart [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import 'dart:typed_data';
+import 'package:meta/meta.dart';
+
+// Described an specific application which can open a file
+class AppHandler {
+  String name;
+  String url;
+  Uint8List? icon;
```
Comment (jagmit) [inline lib/src/app_handler.dart]: We should add short doc comments for each property here

File: lib/src/channel.dart [hunk 5]
```diff
@@ -0,0 +1,101 @@
+import 'dart:async';
+import 'package:flutter/services.dart';
+import 'package:native_file_explorer_launcher/native_file_explorer_launcher.dart';
+
+/// Communication channel between the Dart- and native-layer
```
Comment (jagmit) [inline lib/src/channel.dart]: I think this comment is meant for line 7

File: lib/src/channel.dart [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    } on PlatformException catch (platformError) {
+      print("${platformError.toString()})");
+    }
+    return result;
+  }
+
+  /// Get a list of supported applications which can open the file decribed by [filePath]
+  ///
+  /// Make sure to pass the absolute path of the file.
+  /// Might throw [PlatformException] when the file extension is not included in the [filePath].
```
Comment (jagmit) [inline lib/src/channel.dart:64]: I know the other methods do it the same way but throwing an [`ArgumentError`](https://api.flutter.dev/flutter/dart-core/ArgumentError-class.html) would be more fitting here IMHO

File: lib/src/channel.dart [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  /// Icons are currently not supported in Windows.
+  ///
+  /// [filePath] is the absolute path of the file
+  ///
+  /// Returns a list of [AppHandler] objects, which includes the name, url and icon data of every application. If no application is found, it returns an empty list.
```
Comment (jagmit) [inline lib/src/channel.dart]:
```suggestion
  /// Returns a list of [AppHandler] objects containing metadata of every supported application. If no application is found, it returns an empty list.
```

File: windows/native_file_explorer_launcher_plugin.cpp [hunk 8]
```diff
@@ -10,58 +12,74 @@
...
+      wcstombs_s(&size, &str[0], str.size() + 1, wstr.c_str(), wstr.size());
+
+      return str;
+  }
+
   // Returns the filePath argument from |method_call| if it is present, otherwise
   // returns an empty string.
-  std::string GetfilePathArgument(const flutter::MethodCall<> &method_call)
-  {
+  std::string GetfilePathArgument(const flutter::MethodCall<> &method_call) {
```
Comment (jagmit) [inline windows/native_file_explorer_launcher_plugin.cpp]:
```suggestion
  std::string GetFilePathArgument(const flutter::MethodCall<> &method_call) {
```

**Review by jagmit — COMMENTED** [review]
@leonx98 Just some small nits, code mostly looks got! Thanks for putting yourself through c++ :D

**Comment by leonx98** [comment]
@jagmit thanks for your hints. Branch is now ready to merge 👍

**Comment by jagmit** [comment]
LGTM!

**Pull request merged**